File: src/main/java/me/elsiff/morefish/condition/BiomeCondition.java
package me.elsiff.morefish.condition;

import org.bukkit.Location;
import org.bukkit.block.Biome;
import org.bukkit.entity.Player;

public class BiomeCondition implements Condition {
    private Biome biome;

    public BiomeCondition(Biome biome) {
        this.biome = biome;
    }

    @Override
    public boolean isSatisfying(Player player) {
        Location loc = player.getLocation();
        return (biome == player.getWorld().getBiome(loc.getBlockX(), loc.getBlockZ()));
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/BiomeCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import org.bukkit.block.Biome
import org.bukkit.entity.Item
import org.bukkit.entity.Player

class BiomeCondition(
    private val biomes: Collection<Biome>
) : FishCondition {
    override fun check(
        caught: Item,
        fisher: Player,
        competition: FishingCompetition
    ): Boolean {
        val x = caught.location.blockX
        val y = caught.location.blockY
        val z = caught.location.blockZ
        return caught.world.getBiome(x, y, z) in biomes
    }
}